### DISCUSSION TITLE
Difference between edge and interior ?

### DISCUSSION THREAD

**Discussion by lucacalcaterra** [skupperproject/skupper#900] [Q&A] [answered]
As subject,probably  i missing basic architecture knoledge, but...what' s the difference ? Thanks !

**Comment by ted-ross** [comment, marked as answer]
Interior routers can be connected to each other in any arbitrary topology.  They will discover the topology and automatically compute the least-cost paths across that topology.  Edge routers only connect to interior routers and do not permit incoming connections (i.e. you can't create a link-token on an edge router).  This means that edges are always "leaf nodes" on the topology.

There is a limit to the number of interiors that you can connect into one network.  That limit is 128 interior routers.

There is no limit to the number of edge routers that can be connected to the interior of the network.

There is work in progress (https://github.com/skupperproject/skupper-router/pull/711) that will allow multiple edge routers in one site to be connected together into a full-mesh.  This will greatly improve scale when redundancy is needed at edge locations.

Your question highlights a deficiency in Skupper's documentation.  We will work to improve this.

**Reply by lucacalcaterra** [reply]
@ted-ross very clear ! Thanks for you precious explanation...

Luca

**Comment by ted-ross** [comment]
Hi Luca,

Interior routers can be connected to each other in any arbitrary topology.
They will discover the topology and automatically compute the least-cost
paths across that topology.  Edge routers only connect to interior routers
and do not permit incoming connections (i.e. you can't create a link-token
on an edge router).  This means that edges are always "leaf nodes" on the
topology.

There is a limit to the number of interiors that you can connect into one
network.  That limit is 128 interior routers.

There is no limit to the number of edge routers that can be connected to
the interior of the network.

There is work in progress (
https://github.com/skupperproject/skupper-router/pull/711) that will allow
multiple edge routers in one site to be connected together into a
full-mesh.  This will greatly improve scale when redundancy is needed at
edge locations.

Your question highlights a deficiency in Skupper's documentation.  We will
work to improve this.

Thanks,
-Ted
